edumap/app/models/code.rb

```diff
@@ -12,35 +12,30 @@ class Code < ActiveRecord::Base
   belongs_to :standard
   has_and_belongs_to_many :lessons
 
-  scope :sorted_by, lambda { |sort_option|
+  scope :with_standard_id, -> (ids) { where(standard_id: ids) }
+  scope :with_created_at_gte, -> (ref_date) { where('created_at >= ?', ref_date) }
```

Comment:
Note this is shared with `lesson`; I wonder if it'd make sense to make add a filtering concern that has this and the basic skeleton of `sorted_by` (specifically the logic for errors and reversing the order)